### DISCUSSION TITLE
How to test with jest that a specific language is fully covered, or not ?

### DISCUSSION THREAD

**Discussion by Aarbel** [i18next/i18next#1848] [Q&A]
Hi i18next team,

is there a way to test that a language fully covers the translation of keys of the base language ?

For example:
```
Base language: EN - 450 keys
Translated Language - ES -  400 keys

Test result: error, ES misses 50 keys compared to EN base language

```

Thanks a lot for your insights.

**Comment by KHOANGUYEN196197** [comment]
Yes, there are several ways to test if a translated language fully covers the translation of keys from the base language. Here are a few approaches you can consider:

Key Comparison: Create a script or program that compares the keys in the base language (EN) with the translated language (ES). You can read the keys from both language files or databases and compare them to identify any missing keys. If the number of keys in the translated language is less than the base language, you can determine the difference and generate a report indicating the missing keys.

Automated Testing: Develop an automated testing framework that checks if the translated language covers all the keys in the base language. This can involve writing test cases that verify the presence of each key in the translated language. The automated tests can be run regularly to ensure that the translations remain up to date and complete.

Localization Tools: There are specialized localization tools available that can help manage translations and perform checks for missing keys. These tools often have features that compare the base language with translated versions, flagging any discrepancies or missing translations. They can provide detailed reports highlighting the missing keys, making it easier to identify and rectify any gaps in the translation.

Manual Review: In addition to automated checks, it's beneficial to have human reviewers go through the translated language to verify the completeness of the translation. Experienced translators or language experts can manually review the translated content, cross-checking it against the base language to ensure nothing is missed.

Using a combination of these methods can help ensure that the translated language fully covers the translation of keys from the base language. It's important to establish a robust localization process to maintain the quality and completeness of translations over time.
